Don't show error invalid global times (#36374)

yatharthbajaj30 opened the pull request:
This PR updates both the server-side and client-side message rendering logic so that invalid global times are displayed as plain text instead of showing an error message.

Previously, when a user entered an invalid time such as:

time:2025-10-22T16:00:00-07:aa00

the message displayed:

Invalid time format: 2025-10-22T16:00:00-07:aa00

Now, the message simply displays:

time:2025-10-22T16:00:00-07:aa00

alya commented:
Thanks! Please go through our [guide on submitting a pull request](https://zulip.readthedocs.io/en/latest/contributing/reviewable-prs.html) to learn how to present your proposed changes to Zulip, so that your work can be reviewed.

yatharthbajaj30 commented:
**Don't show error for invalid global times**

This PR updates both the frontend and backend logic to ensure that invalid global time inputs are displayed as plain text instead of showing an error message.

Overview of changes

Frontend:
Removed the time-error <span> element that previously displayed an error message when encountering an invalid time format.
This change was necessary because once a message is fetched from the backend or rendered dynamically, updating the error span was difficult and inconsistent with Zulip’s rendering flow.

Backend (__init__.py):
Updated the message-handling logic to modify how invalid time strings are returned from the backend.
Instead of sending an error message, the backend now returns the original text as-is, allowing it to render normally on the client.

Behavior before

For an invalid time input such as:

time:2025-10-22T16:00:00-07:aa00


Zulip displayed:

Invalid time format: 2025-10-22T16:00:00-07:aa00

Behavior now

The same message now renders as plain text:

time:2025-10-22T16:00:00-07:aa00

**Screenshots**
<details> <summary>Before</summary>
<img width="2411" height="1340" alt="Screenshot 2025-10-25 151137" src="https://github.com/user-attachments/assets/a84bccec-549f-4af1-972e-705118599c12" />


</details> <details> <summary>After</summary>
<img width="1389" height="1037" alt="Screenshot 2025-10-25 174208" src="https://github.com/user-attachments/assets/d9d83589-7548-4716-8513-0f179dcbe377" />


</details>
**_Notes_**

The frontend modification improves rendering consistency and prevents unnecessary error elements in the message DOM.

The backend change ensures that invalid times no longer interrupt message rendering or display an unnecessary warning to users.****
